nativelib/src/main/scala/scala/scalanative/runtime/ieee754tostring/ryu/RyuDouble.scala

```diff
@@ -699,21 +699,80 @@ object RyuDouble {
       roundingMode: RyuRoundingMode
   ): String = {
 
-    // Step 1: Decode the floating point number, and unify normalized and
-    // subnormal cases.
-    // First, handle all the trivial cases.
+    // Handle all the trivial cases.
     if (value.isNaN) return "NaN"
     if (value == Double.PositiveInfinity) return "Infinity"
     if (value == Double.NegativeInfinity) return "-Infinity"
     val bits = java.lang.Double.doubleToLongBits(value)
     if (bits == 0) return "0.0"
     if (bits == 0x8000000000000000L) return "-0.0"
 
-    // Otherwise extract the mantissa and exponent bits and run the full
-    // algorithm.
+    val result = new scala.Array[Char](RyuDouble.RESULT_STRING_MAX_LENGTH)
+    val strLen = doubleToChars(value, roundingMode, result, 0)
+
+    new String(result, 0, strLen)
+  }
+
+  @inline
```

Comment:
It does not affect the central idea of this PR.

Again, depending upon estimates of usage and personal preference, I would try without the "@inline".  NaN & kin need to be correct but not necessarily lightening fast.  People get concerned about code size.  Although below I am going to suggest something which will increase code size.

Reply:
| I thought that since it's private + only two lines of code, it should be totally fine. Really against this?
I am not really 'against` this and I do not want to hold this PR up.

As a fellow devo, I am leery that it is doing what you intend.  I am also aware that my wanting to save
a few bytes is silly given the HUGE data tables declared in this file.

My working assumption is that the compiler can do  better at figuring out when to inline than 
I can on-the-fly. I am not sure how good that part of the SN compiler is but GCC and LLVM are
reputed to be excellent & getting better.

When something is inline'ed on an `if` path, it increases the size of the gap that the 'branch-not-taken' 
must jump over.  I believe that on modern machines, both paths are fetched onto the instruction (I-)cache and 
may be speculatively executed.  Inline code takes up more of the I-cache.

If these were 'frequent" paths, as an analyst, I would expect `@inline` to have a (small) actual
benefit (but would still try to measure). Since they are 'cold' paths I doubt they are worth it.

However, you may prefer a different end of the egg.

Reply:
> def this(start: Int, length: Int, data: Array[Char]) = {
Indeed, that would save I think both an Array allocation and the arraycopy you mentioned.
Good catch!

You/we would need to change the scope of the called constructor away from `private[lang]`
to something like `private[scalanative]` or `private[scala.scalanative]`  We would have to try 
one or both to see if/what works. I think that having the constructor public is not an option.

Reply:
>You/we would need to change the scope of the called constructor away from private[lang]
to something like private[scalanative] or private[scala.scalanative]

Good point. I fear it won't be possible to match both namespaces (`scala.scalanative` and `javalib`), so it's quite an issue.

Another option would be to just remove the entry points `floatToString()/doubleToString()`.
This would also solve the previous point about `_xxxToCharsNoCheck()`.

I remember you rose this concern in the issue #2902:
_The SN implementation was announced a long time on the Ryu repository page as existing. I know of no direct users
but, I believe, we must behave as though at least one exists.
That means preserving the current method but also does not mean that an overload is not possible/feasible._

An other alternative could be to leave this code as is (non optimized) and to mark the `floatToString()/doubleToString()` methods as  `@deprecated`.

Reply:
> If not I would vote for removing the xxToString methods in the 0.5 branch.

That is certainly one way to find out if anyone is using them ;-)  

It would also solve a lot of finagling that you/we have been doing in this code.

Worst that could happen is that we have to re-introduce the calls in a year.

If you decide to go down this path, there are some documentation implications.
Normally, I like to see code, tests, and documentation all in the same PR.
I propose  getting the code & tests clean, slick, & merged and
creating another Issue stating that the 0.5.0 Release Notes need a section
which:
1) Describes the removal of the d2s and f2s methods, the reason why and the
    suggested alternative.

2) Gives an example, probably for d2s, off a very thin wrapper for d2c which 
    unconditionally allocates an Array (and showing the new constant to use)
    calls d2c, and creates the string.

That keeps this PR from getting delayed for long periods waiting for a lock
on the 0.5.0 Release Notes.  It also allows us to gain some experience.
If we have to back the change out, it means that we will not have spent
time crafting documentation which becomes obsolete.

I mean to sound judicious about merge process issues, not grouchy or frustrated.
You have put so much time and detail into this, you are the one who has a right to feel frustrated.

PS:  I am _so_ glad not to have that private String constructor visible to "package[scala]".

Reply:
Actually, my favorite way would be to have a @deprecated annotation that is not breaking compilation...

We could also implement alternative wrappers like:

```scala
  def doubleToChars(
      value: Double,
      roundingMode: RyuRoundingMode
  ): (scala.Array[Char], Int) = { ... }
```

or

```scala
  def doubleToString(
      value: Double,
      roundingMode: RyuRoundingMode,
      createStringFn: (scala.Array[Char], Int) => String
  ): String = { ... }
```

Reply:
slightly off topic but for amusement, what errors where you getting when you tried "@Deprecated"

Reply:
>Deleting the d2s and f2s routines, until we can no longer get away with it, vastly simplifies the implementation of this PR as you suggested.

Ok so let's do that now.

>what errors where you getting when you tried "@deprecated"

```
[error] ./scala-native/javalib/src/main/scala/java/lang/Float.scala:323:14: method floatToString in object RyuFloat is deprecated: Use floatToChars() instead, as it requires no additional String/Array allocations
[error]     RyuFloat.floatToString(f, RyuRoundingMode.Conservative)
[error]              ^
[error] one error found
[error] (javalib2_11 / Compile / compileIncremental) Compilation failed
```